File: src/tools/shell.ts
import { Tool } from './tool';
import { registerTool } from './registry';
import { exec } from 'child_process';

export const shell: Tool = {
  name: 'shell',
  description: 'Execute shell commands',
  schema: { type: 'string', description: 'Shell command to execute' },
  handler: (cmd: string) => new Promise((resolve, reject) => {
    exec(cmd, (err, stdout, stderr) => {
      if (err) {
        reject(stderr || err.message);
      } else {
        resolve(stdout);
      }
    });
  })
};

registerTool(shell);

File: src-tauri/src/tools/shell.rs
use super::{Tool, ToolMetadata};
use std::process::Command;

pub struct Shell;

impl Tool for Shell {
    fn metadata(&self) -> ToolMetadata {
        ToolMetadata {
            name: "shell",
            description: "Execute shell commands",
        }
    }

    fn handle(&self, input: String) -> anyhow::Result<String> {
        let output = Command::new("sh").arg("-c").arg(input).output()?;
        Ok(String::from_utf8_lossy(&output.stdout).to_string())
    }
}
